### PR TITLE
WIP Hide empty submodules button

### LINKED ISSUE #5704: Hide submodules toolbar icon if no submodules exist
**Do you want to request a *feature* or report a *bug*?**

UX issue / bug

**What is the current behavior?**

For a repo with no submodules, the toolbar shows this:

image

Neither of these submenu items do anything useful. Both are confusing.

**What is the expected behavior?**

Currently this toolbar item has no value other than to cause confusion and clutter.

It should either be made useful, or removed. Submodules are a relatively "advanced" and rarely used git feature, and this button has prime position for something that's ultimately useless if you're not using submodules.

I favour removal&mdash;anything we can do to clean up the UI is a win.

If keeping it, the button should not have child items (no submodules) and should just open the "Manage Submodules" UI directly.

**Environment you encounter the issue:**

- GitExtensions version: 3.0 rc 1

### PR DESCRIPTION
**Pull request by drewnoakes**
Fixes #5704

## Changes
- Hide submodule toolbar button when not needed
- Show the 'navigate upwards' icon when there is a supermodule.
- Change the submodules toolbar icon to make it less intense
 
#### Has submodules:
image

#### No submodules:
image

#### Is a submodule (has a supermodule):
image

---

#### Has submodules:
image

#### No submodules:
image

#### Is a submodule (has a supermodule):
image

## Quality
- Manual testing

## Testing
- GIT 2.19.1
- Windows 10

### CODE REVIEW

File: GitUI/CommandsDialogs/FormBrowse.cs [hunk 1]
```diff
@@ -313,7 +313,20 @@ public FormBrowse([NotNull] GitUICommands commands, string filter, ObjectId sele
...
+            toolStripButtonLevelUp.Visible = false;
             _submoduleStatusProvider = new SubmoduleStatusProvider();
+            _submoduleStatusProvider.SubmodulesChanged += (sender, e) =>
+            {
+                ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+                {
+                    await this.SwitchToMainThreadAsync();
+
+                    toolStripButtonLevelUp.Visible = e.HasSubmodules || e.HasSupermodule;
+                    toolStripButtonLevelUp.Image = e.HasSupermodule ? Images.NavigateUp : Images.SubmodulesManage;
```
Comment (gerhardol) [inline GitUI/CommandsDialogs/FormBrowse.cs]: Images.SubmodulesManage I prefer Images.FolderSubmodule in this context, hard to see Sucmodule manage here.
Images.NavigateUp Is this better? Hesitant
None Hesitant

What to use in the documentation?

Comment (drewnoakes) [reply GitUI/CommandsDialogs/FormBrowse.cs]: On the icon choice, I find the icon with the blue background too visually dominant relative to the other toolbar icons. Is it really too hard to see now, or would it just be a matter of getting used to the change? There may be a middle ground. Interested to hear other people's opinions too. 

Regarding the icon change, my concern is that the button sometimes navigates upwards, sometimes opens the submodule. It's not clear from the icon when to expect either behaviour. I would also be happy to have 'navigate to supermodule' become a submenu item.

> What to use in the documentation?

I don't follow, sorry.

Comment (gerhardol) [reply GitUI/CommandsDialogs/FormBrowse.cs]: > Is it really too hard to see now, or would it just be a matter of getting used to the change? 

I know where to click anyway and can get used to it. Just voting against the change.

> I don't follow, sorry.

When we update the documentation, contextual need to be described somehow.
To answer this myself: Use "submodule" icon in the screenshot, describe the uparrow and removal.

It is better to merge PRs like this before anyone start to update the documentation.

Comment (drewnoakes) [reply GitUI/CommandsDialogs/FormBrowse.cs]: Thanks for clarifying around the docs. Sounds like we need a cutoff for changes like this, so we can get the translation and documentation phases underway without wasting effort. If this PR doesn't make the bar, I can retarget it to a 3.1 branch or similar (or maybe we make a 3.0 branch and keep master for 3.1).

**Comment by RussKie** [comment]
I share the sentiment:
* it is the prime realestate, and if nothing to show - don't;
* I too find the current icon a bit to overpowering.

Beyond this I haven't had much time to give it a deeper thought. 
However I'd downgrade the icon for v3. For the other items, would it be ok to park it until after v3 is out, so we can find a solution that works for all?

@gerhardol  If/when submodules move to the left panel, would we need to keep them in the toolbar?

File: GitUI/CommandsDialogs/FormBrowse.cs (hunk 1 above)
Comment (RussKie) [reply GitUI/CommandsDialogs/FormBrowse.cs]: I would avoid multiple branches, if possible. Reconciliation can be nightmarish.
v3 is not far away, after that we should be back to normal.

**Comment by amaiorano** [comment]
@drewnoakes 
> Regarding the icon change, my concern is that the button sometimes navigates upwards, sometimes opens the submodule. It's not clear from the icon when to expect either behaviour. I would also be happy to have 'navigate to supermodule' become a submenu item.

Although you're right that it's inconsistent, I think it's important that clicking on the button goes to the super project. It's faster for navigation. What you did here to change the icon will help a great deal in understanding when clicking will go to super project.

Personally, I like this change.

**Comment by gerhardol** [comment]
I suggest we take a decision on this before 3.0, someone should update the documentation soon

**Comment by drewnoakes** [comment]
Fixed conflict.

**Comment by iTob191** [comment]
I really like the proposed changes!

But in case there are submodules but no superproject, I would display a different tooltip on the button. Something like "Manage submodules" or simply "Submodules...".

**Comment by drewnoakes** [comment]
I would like to revisit this and split the "up to superproject" function out into its own button. In such cases we'd see both the blue circular up-arrow and the submarine icon.

**Comment by RussKie** [comment]
@drewnoakes @gerhardol what is the plan for this one?

**Comment by gerhardol** [comment]
I have already been voted down so merge.

But please reconsider changing the icon for top module.
I can rebase if Drew is still stuck between a rock and a hard place.

**Comment by RussKie** [comment]
I don't have particular strong feelings either way, and happy to defer the change until after a v3.1 release

**Comment by gerhardol** [comment]
Move it to 3.2.0 then

**Comment by drewnoakes** [comment]
> I can rebase if Drew is still stuck between a rock and a hard place.

Feel free. I've spent the last months preparing for then executing the move of my family halfway around the world. Finally here and getting settled, but have a backlog of paid work I need to get on with and don't know when I'll have time to look at this.

**Comment by gerhardol** [comment]
Closing in favor of #7770 
This PR has a few other changes, some are solved differently in master

**Pull request closed without merging**